gadget/layout.go

```diff
@@ -328,14 +311,8 @@ func LayoutVolume(volume *Volume, opts *LayoutOptions) (*LaidOutVolume, error) {
 		}
 	}
 
-	volumeSize := quantity.Size(farthestEnd)
-	if fartherstOffsetWrite+quantity.Offset(SizeLBA48Pointer) > farthestEnd {
-		volumeSize = quantity.Size(fartherstOffsetWrite) + SizeLBA48Pointer
-	}
-
```

Comment:
doesn't this logic need to apply to some other of the sizes now?

Reply:
`EnsureVolumeCompatibility` checks `OnDiskVolume.UsableSectorsEnd` against `gadget.Volume.MinSize()`, while in the past was using `LaidOutVolume.Size`. So in fact this calculation should be included in `MinSize()`, so the (very unusual?) case of wanting to write the offset of the structure somewhere out of the defined structures is taken into account.

This made me look at usage of `{LaidOutStructure,LaidOutContent}.AbsoluteOffsetWrite` and apparently they are not used anywhere, not even in `ubuntu-image`. And `VolumeContent.OffsetWrite` is not used either. The only related thing that is used is `VolumeStructure.OffsetWrite`, by `ubuntu-image`, here:
https://github.com/canonical/ubuntu-image/blob/main/internal/statemachine/helper.go#L471
and in that case it is not even considering `RelativeOffset.RelativeTo`, so it uses the relative offset as absolute. It works for the mbr because even though we have `offset-write: mbr+92`, the mbr structure has anyway offset 0. So basically it works by chance.

We probably need to discuss what to do with this.